### PR TITLE
upload to Google Play with GitHub Actions, remove Bitrise

### PR DESCRIPTION
**Pull request by connyduck**
closes https://github.com/tuskyapp/Tusky/issues/4236

I hope I did everything correctly, we will find out after this is merged.

I am setting up the environment variables containing the keys so they are only available to actions that need them.

This should be easier to maintain, GitHub seems to be faster then Bitrise, and we no longer have two checks on every pull request.

### CODE REVIEW

**Comment by charlag** [comment]
Looks good so far!

File: .github/workflows/check-and-build.yml [hunk 1]
```diff
@@ -1,15 +1,11 @@
-name: CI
+name: Check and build
 
 on:
-  push:
-    tags:
-      - '*'
   pull_request:
```
Comment (Goooler) [inline .github/workflows/check-and-build.yml:4]:
```suggestion
  push:
    branches:
      - develop
  pull_request:
```

File: .github/workflows/check-and-build.yml [hunk 2]
```diff
@@ -20,14 +16,11 @@ jobs:
...
-        uses: gradle/actions/wrapper-validation@v4
-
       - name: Copy CI gradle.properties
         run: mkdir -p ~/.gradle ; cp .github/ci-gradle.properties ~/.gradle/gradle.properties
 
       - name: Gradle Build Action
-        uses: gradle/gradle-build-action@v3
+        uses: gradle/actions/setup-gradle@v4
         with:
           cache-read-only: ${{ github.ref != 'refs/heads/main' && github.ref != 'refs/heads/develop' }}
```
Comment (Goooler) [inline .github/workflows/check-and-build.yml:25]:
```suggestion
```

File: .github/workflows/deploy-release.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+      - uses: r0adkll/sign-android-release@f30bdd30588842ac76044ecdbd4b6d0e3e813478
+        name: Sign Tusky Blue aab
+        id: sign_aab
+        with:
+          releaseDirectory: app/build/outputs/bundle/blueRelease
+          signingKeyBase64: ${{ secrets.KEYSTORE }}
+          alias: ${{ secrets.KEY_ALIAS }}
+          keyStorePassword: ${{ secrets.KEYSTORE_PASSWORD }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
```
Comment (Goooler) [inline .github/workflows/deploy-release.yml:27]: We can replace this step with:

https://github.com/LawnchairLauncher/lawnchair/blob/2d86033f7a1f0e3fd32ba395118d80e73832c2d3/.github/workflows/ci.yml#L44-L53

https://github.com/LawnchairLauncher/lawnchair/blob/2d86033f7a1f0e3fd32ba395118d80e73832c2d3/build.gradle#L194-L210

Comment (connyduck) [reply .github/workflows/deploy-release.yml:27]: I'd prefer it if we don't touch the build.gradle files

File: .github/workflows/check-and-build.yml (hunk 1 above)
Comment (connyduck) [reply .github/workflows/check-and-build.yml:4]: Why? The action already that runs on develop already includes this action, this would run it twice?

Comment (Goooler) [reply .github/workflows/check-and-build.yml:4]: You mean `.github/workflows/populate-gradle-build-cache.yml`? It has been deleted in this PR.

File: .github/workflows/deploy-release.yml (hunk 3 above)
Comment (Goooler) [reply .github/workflows/deploy-release.yml:27]: I'll address that in a separated PR.

File: .github/workflows/check-and-build.yml (hunk 1 above)
Comment (connyduck) [reply .github/workflows/check-and-build.yml:4]: No, I meant the `deploy-test` has `uses: ./.github/workflows/check-and-build.yml` and runs on develop, but it actually doesn't 😅 
I'll fix that, thanks

**Review by Tak — APPROVED** [review]
Looks reasonable to me? (I'm neither an expert in github actions nor in publishing to play store)

**Comment by connyduck** [comment]
Yeah, neither am I, I fully expect some kind of mistake but we will only know when we try it out.

**Pull request merged**